### PR TITLE
Fix AL crash with ME hatches

### PR DESCRIPTION
**Pull request by HoleFish**
For example:
image
image
Result:
image
This pr tries to fix it.

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java [hunk 1]
```diff
@@ -369,19 +374,160 @@ public CheckRecipeResult checkProcessing() {
...
+    public static void consumeInputItems(ArrayList<GT_MetaTileEntity_Hatch_InputBus> inputBusses, int amountMultiplier,
+        int[] itemConsumptions, Map<GT_Utility.ItemId, ItemStack> inputsFromME) {
+
+        for (int i = 0; i < itemConsumptions.length; i++) {
+            GT_MetaTileEntity_Hatch_InputBus inputBus = inputBusses.get(i);
+            if (!inputBus.isValid()) continue;
+            ItemStack item;
+            if (inputBus instanceof GT_MetaTileEntity_Hatch_InputBus_ME meBus) {
+                item = meBus.getStackInSlot(0);
+                if (item != null) {
```
Comment (Laiff) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java:501]: This check broke rule of basic AL logic: items should be stored in right slot. Can you add inputs order in ME cache and try to consume it from only one index not from whole map?

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java [hunk 2]
```diff
@@ -369,19 +374,160 @@ public CheckRecipeResult checkProcessing() {
...
+
+    public static void consumeInputFluids(ArrayList<GT_MetaTileEntity_Hatch_Input> inputHatches, int amountMultiplier,
+        FluidStack[] fluidConsumptions, Map<Fluid, FluidStack> fluidsFromME) {
+        for (int i = 0; i < fluidConsumptions.length; i++) {
+            GT_MetaTileEntity_Hatch_Input inputHatch = inputHatches.get(i);
+            if (!inputHatch.isValid()) continue;
+            FluidStack fluid;
+            if (inputHatch instanceof GT_MetaTileEntity_Hatch_Input_ME meHatch) {
+                fluid = meHatch.getMatchingFluidStack(fluidConsumptions[i]);
+                if (fluid != null) {
```
Comment (Laiff) [inline src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java:522]: Same here

**Review by Laiff — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java (hunk 2 above)
Comment (HoleFish) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java:522]: Fluids have never had such a rule. Their only requirement is to be in seperate hatches.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java (hunk 1 above)
Comment (HoleFish) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java:501]: ME cache map has no order and should not have. ME bus just checks the item in the first stocking slot, pulls it from net storage (the map) and consumes it.

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java (hunk 2 above)
Comment (Laiff) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java:522]: It is same rule

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java (hunk 1 above)
Comment (Laiff) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java:501]: Ok it also can works

File: src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java (hunk 2 above)
Comment (HoleFish) [reply src/main/java/gregtech/common/tileentities/machines/multi/GT_MetaTileEntity_AssemblyLine.java:522]: not the rule the old code implied. ok I will fix it if you want

**Comment by HoleFish** [comment]
Should be careful. Have been suffering from ME input buses/hatches.

**Review by Laiff — APPROVED** [review]

**Pull request merged**